river/command/keyboard.zig

```diff
@@ -58,15 +72,33 @@ pub fn keyboardLayout(
     ) orelse return error.InvalidValue;
     defer new_keymap.unref();
 
-    server.config.keymap.unref();
-    server.config.keymap = new_keymap.ref();
+    try applyLayout(new_keymap);
+}
 
-    var it = server.input_manager.devices.iterator(.forward);
-    while (it.next()) |device| {
-        if (device.wlr_device.type != .keyboard) continue;
-        const wlr_keyboard = device.wlr_device.toKeyboard();
-        // wlroots will log an error if this fails and there's unfortunately
-        // nothing we can really do in the case of failure.
-        _ = wlr_keyboard.setKeymap(new_keymap);
-    }
+pub fn keyboardLayoutFile(
+    _: *Seat,
+    args: []const [:0]const u8,
+    _: *?[]const u8,
+) Error!void {
+    var file = std.fs.cwd().openFile(args[1], .{}) catch return error.CannotReadFile;
+    defer file.close();
+
+    var file_bytes = file.readToEndAlloc(util.gpa, 512 * 4096) catch |err| {
```

Comment:
Not a fan of uncommented magic numbers in code, TBH.

Reply:
Well, I stole that one from here:
https://github.com/ifreund/zig-wayland/blob/7f528883e057b282b1cb51db2899f9183b2e5a3b/src/scanner.zig#L191

It's a huge overkill for this purpose, but I don't know what's the best way to come up with a reasonable estimated upper limit for the size of XKB files. :thinking:

Reply:
The size isn't the issue, just that it is arbitrary. `readToEndAlloc()` will only allocate as much as is needed to store the contents of the file. As such, I'd use `math.maxInt(usize)`.

And one more thing I forgot on the first review round: `readToEndAlloc()` can return `error.FileTooBig`, which should make your function return `error.OutOfMemory`.